### PR TITLE
Show transcript for lectures with subtitles

### LINKED ISSUE #1369: Show transcript for lectures with subtitles
**Is your feature request related to a problem? Please describe.**
\-

**Describe the solution you'd like**
- Add the option to see a searchable transcript next to the lecture video if subtitles are available.
- Transcript should be synchronized with the current progress of the video (and, for example, also show the current video timestamp - see example screenshot from YouTube below).

**Describe alternatives you've considered**
Download option for transcripts as a text file.

**Additional context**
1. I'd work on this issue as part of the Open Source Lab

<details>
<summary>2. YouTube transcript function</summary>  

image

</details>

### PR DESCRIPTION
**Pull request by carlobortolan**
### Motivation and Context
See #1369.

### Description
Updated web client to include the option to show lecture transcript with available subtitles.
When the transcript is enabled, the user can read/scroll/fast-forward through the text (see demo video below).

### Steps for Testing
Prerequisites:
- 1 Video with subtitles
- 1 Video without subtitles

1. Open a video.
2. a "Show transcript" button should appear if it has subtitles.
3. Now, you should be able to see the transcript next to the video (/ below the video if you're on a mobile device). 

### Screenshots
https://github.com/user-attachments/assets/912302be-2ae6-4f56-8c07-3760f80e0991

### CODE REVIEW

File: web/template/watch.gohtml [hunk 1]
```diff
@@ -424,6 +424,22 @@
             </div>
         {{end}}
 
+        <!-- Transcript -->
+        <div id="transcript-desktop" x-cloak="" x-show="sidebar === watch.SidebarState.Transcript && transcriptAvailable"
```
Comment (joschahenningsen) [inline web/template/watch.gohtml]:
```suggestion
        <div id="transcript-desktop" x-cloak x-show="sidebar === watch.SidebarState.Transcript && transcriptAvailable"
```

File: web/template/partial/stream/transcript.gohtml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    <span class="text-xs font-semibold text-white bg-red-500 rounded-full px-2 py-1 italic">beta</span>
+                </div>
+                <button type="button" title="Download transcript file"
+                        class="text-3 text-xs md:text-sm font-semibold hover:bg-gray-100 dark:hover:bg-gray-600 rounded-full px-2 py-1"
+                        @click="transcriptController.downloadTranscript()">
+                    Export transcript
+                </button>
+            </div>
+            <div class="flex-grow overflow-hidden relative">
+                {{template "transcript-list" .}}
```
Comment (joschahenningsen) [inline web/template/partial/stream/transcript.gohtml]: No need to pass any context, right?

```suggestion
                {{template "transcript-list"}}
```

File: web/template/watch.gohtml [hunk 3]
```diff
@@ -475,14 +491,21 @@
...
+                                    class="rounded-lg px-3 py-1 md:px-4 py-2 h-fit w-fit bg-gray-100 hover:bg-gray-200 dark:bg-secondary-light dark:hover:bg-gray-600"
                                     :title="'New Bookmark'">
                                 <i class="fa-solid fa-bookmark text-4"></i>
                             </button>
                         {{end}}
 
+                        <!-- Transcript Button -->
+                        <button x-show="transcriptAvailable" @toggletranscript.window="e => {transcriptAvailable=true}" @click="sidebar = (sidebar === watch.SidebarState.Transcript ? watch.SidebarState.Hidden : watch.SidebarState.Transcript)"
+                                class="rounded-lg px-3 py-1 md:px-4 py-2 h-fit w-fit bg-gray-100 hover:bg-gray-200 dark:bg-secondary-light dark:hover:bg-gray-600"
+                                :title="'Show Transcript'">
```
Comment (joschahenningsen) [inline web/template/watch.gohtml]: This way we don't need to run this string through javascript
```suggestion
                                title="Show Transcript">
```

Comment (joschahenningsen) [reply web/template/watch.gohtml]: Seems like we made this mistake before in these files, feel free to clean them up if you want, if not no worries.

File: web/ts/watch.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
 import { getPlayers } from "./TUMLiveVjs";
 import { copyToClipboard, Time } from "./global";
-import { seekbarOverlay } from "./seekbar-overlay";
```
Comment (joschahenningsen) [inline web/ts/watch.ts:3]: I assume this was unused?

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
Very clean, like it a lot! Just left a few nits.

File: web/template/watch.gohtml (hunk 1 above)
Comment (carlobortolan) [reply web/template/watch.gohtml]: Looks like a copy-paste error like the one related to the unnecessary `'` :sweat_smile:  
image

I fixed it in my latest commit.

File: web/template/partial/stream/transcript.gohtml (hunk 2 above)
Comment (carlobortolan) [reply web/template/partial/stream/transcript.gohtml]: :+1:

File: web/template/watch.gohtml (hunk 3 above)
Comment (carlobortolan) [reply web/template/watch.gohtml]: Thanks for pointing that out; I removed all the unnecessary `'` I could find in my latest commit.

File: web/ts/watch.ts (hunk 4 above)
Comment (carlobortolan) [reply web/ts/watch.ts:3]: Yes (see beginning & end of [watch.ts file](https://github.com/TUM-Dev/gocast/blob/dev/web/ts/watch.ts))

**Review by joschahenningsen — APPROVED** [review]
🚀

**Pull request merged**